sos/report/plugins/maas.py

```diff
@@ -67,58 +62,59 @@ def _is_snap_installed(self):
             return maas_pkg['pkg_manager'] == 'snap'
         return False
 
+    def _snap_collect(self):
+        self.add_cmd_output([
+            'snap info maas',
+            'maas status'
+        ])
+
+        self.add_forbidden_path([
+            "/var/snap/maas/**/*.key",
+            "/var/snap/maas/**/*.pem",
+            "/var/snap/maas/**/secret",
+        ])
+
+        self.add_copy_spec([
+            "/var/snap/maas/common/log",
+            "/var/snap/maas/common/pebble/.pebble.state",
+            "/var/snap/maas/common/snap_mode",
+            "/var/snap/maas/current/**/*.conf",
+            "/var/snap/maas/current/**/*.yaml",
+            "/var/snap/maas/current/bind",
+            "/var/snap/maas/current/preseeds",
+            "/var/snap/maas/current/supervisord",
+        ])
+
+    def _deb_collect(self):
+        self.add_cmd_output([
+            "apt-cache policy maas maas-*",
+        ])
+
+        self.add_forbidden_path([
+            "/var/lib/maas/**/*.key",
+            "/var/lib/maas/**/*.pem",
+            "/var/lib/maas/**/secret",
+            "/etc/maas/**/*.key",
+            "/etc/maas/**/*.pem",
+            "/etc/maas/**/secret",
+        ])
+
+        self.add_copy_spec([
+            "/etc/maas/**/*.conf",
+            "/etc/maas/**/*.yaml",
+            "/etc/maas/preseeds",
+            "/var/lib/maas/**/*.conf",
+            "/var/lib/maas/dhcp",
+            "/var/lib/maas/rsyslog",
+            "/var/lib/maas/temporal",
+            "/var/log/maas",
```

Comment:
with the improvements we've been discussing, is it worth splitting this apart so that we collect recent logs normally, and we collect all when we specify `--all-logs`

Reply:
Thanks for the feedback! 

So we always want to copy recent `.log` files, but _rotated logs_ (e.g. `.log.1.gz`) only when `--all-logs` is provided?
 
```python
if self.get_option("all_logs"):
    self.add_copy_spec([
        "/var/log/maas/",
    ])
else:
    self.add_copy_spec([
        "/var/log/maas/*.log",
    ])
```

Reply:
I don't have a deb implementation myself, but something like the following would work for the snap version

```
if self.get_option("all_logs"):
    self.add_copy_spec([
        "/var/snap/maas/common/log/",
    ])
else:
    self.add_copy_spec([
        "/var/snap/maas/common/log/*.log",
        "/var/snap/maas/common/log/*/*.log",
    ])
```

We may need to look at the `rsyslog` dir, on how best to accommodate that as that has folders already with date timestamps. Depending on the number of nodes in the environment, having all of the logs for a non `--all-logs` could be overwhelming. If we could somehow get the latest directory from that, then that would work in the `else` section. Thoughts? You're the MAAS expert, I let you decide :)

Reply:
Added some logic to handle `rsyslog`. However it is applicable only to MAAS before 3.5 (as in 3.5 we keep everything in the journal)

Reply:
ack, sounds good to me